src/marqo/core/search/hybrid_search.py

```diff
@@ -0,0 +1,212 @@
+            limit=result_count,
+            offset=offset, showHighlights=False, filter=filter_string, attributesToRetrieve=attributes_to_retrieve,
+            boost=boost, image_download_headers=image_download_headers, context=context, scoreModifiers=score_modifiers,
+            index=marqo_index, modelAuth=model_auth, text_query_prefix=text_query_prefix,
+            hybridParameters=hybrid_parameters
+        )]
+
+        with RequestMetricsStore.for_request().time(f"search.hybrid.vector_inference_full_pipeline"):
+            qidx_to_vectors: Dict[Qidx, List[float]] = run_vectorise_pipeline(config, queries, device)
+        vectorised_text = list(qidx_to_vectors.values())[0]
+
+        # Parse text into required and optional terms.
+        (required_terms, optional_terms) = utils.parse_lexical_query(query_text_search)
+
+        marqo_query = MarqoHybridQuery(
+            index_name=index_name,
+            vector_query=vectorised_text,
+            filter=filter_string,
+            limit=result_count,
+            ef_search=ef_search,
+            approximate=approximate,
+            offset=offset,
+            or_phrases=optional_terms,
+            and_phrases=required_terms,
+            attributes_to_retrieve=attributes_to_retrieve,
+            searchable_attributes=searchable_attributes,
+            score_modifiers=score_modifiers.to_marqo_score_modifiers() if score_modifiers is not None else None,
+            # Hybrid-specific attributes
+            score_modifiers_lexical=hybrid_parameters.score_modifiers_lexical.to_marqo_score_modifiers()
+            if hybrid_parameters.score_modifiers_lexical is not None else None,
+            score_modifiers_tensor=hybrid_parameters.score_modifiers_tensor.to_marqo_score_modifiers()
+            if hybrid_parameters.score_modifiers_tensor is not None else None,
+            hybrid_parameters=hybrid_parameters
+        )
+
+        vespa_index = vespa_index_factory(marqo_index)
+        vespa_query = vespa_index.to_vespa_query(marqo_query)
+
+        total_preprocess_time = RequestMetricsStore.for_request().stop("search.hybrid.processing_before_vespa")
+        logger.debug(
+            f"search (hybrid) pre-processing: took {(total_preprocess_time):.3f}ms to vectorize and process query.")
+
+        # SEARCH TIMER-LOGGER (roundtrip)
+        with RequestMetricsStore.for_request().time("search.hybrid.vespa",
+                                                    lambda t: logger.debug(f"Vespa search: took {t:.3f}ms")
+                                                    ):
+            try:
+                responses = config.vespa_client.query(**vespa_query)
+            except VespaStatusError as e:
+                # The index will not have the embedding_similarity rank profile if there are no tensor fields
+                if f"No profile named '{RANK_PROFILE_HYBRID_CUSTOM_SEARCHER}'" in e.message:
```

Comment:
Is this the only error we'd like to handle from the vespa response?

Reply:
It's probably not, but it's the only one i found that's relevant right now. We can continue adding others in future features. This error will be quite common for people who try to hybrid search without having both a lexical and a tensor field